userapi/src/main/java/fk/prof/userapi/cache/ClusteredProfileCache.java

```diff
@@ -0,0 +1,281 @@
+
+                // able to claim the ownership. start loading it.
+                Future<AggregatedProfileInfo> profile = loadProfile(profileName);
+                if (profile.isComplete()) {
+                    if(profile.succeeded()) return profile.result();
+                    else throw new CachedProfileNotFoundException(profile.cause());
+                }
+
+                throw new ProfileLoadInProgressException(profileName);    //fail because loading is still not complete
+            }
+        });
+    }
+
+    /**
+     * Main method to get the already cached view / create view for the cached profile. If the profile is not cached,
+     * the return future will fail according to {@code getAggregatedProfile}.
+     *
+     * @param profileName name of the profile
+     * @param traceName name of the trace
+     * @param profileViewType type of profileView
+     * @return Future containing a pair of trace specific aggregated samples and its view.
+     */
+    public <T extends ProfileView>
+    Future<Pair<AggregatedSamplesPerTraceCtx, T>> getProfileView(AggregatedProfileNamingStrategy profileName,
+                                                                 String traceName, ProfileViewType profileViewType) {
+        Future<Pair<AggregatedSamplesPerTraceCtx, T >> viewFuture = Future.future();
+        Pair<Future<AggregatedProfileInfo>, Cacheable<ProfileView>> profileViewPair = cache.getView(profileName, traceName, profileViewType);
+
+        if (profileViewPair.first != null) {
+            if (profileViewPair.second != null) {
+                return Future.succeededFuture(
+                    Pair.of(profileViewPair.first.result().getAggregatedSamples(traceName), (T) profileViewPair.second));
+            }
+
+            return doAsync(() -> getOrCreateView(profileName, traceName, profileViewType));
+        }
+
+        // profile not found, so initiate the profile loading and compute view after that
+        getAggregatedProfile(profileName).setHandler(ar -> {
+            if(ar.succeeded()) {
+                try {
+                    viewFuture.complete(getOrCreateView(profileName, traceName, profileViewType));
+                    return;
+                } catch (Exception e) {
+                    viewFuture.fail(e);
+                    return;
+                }
+            }
+            viewFuture.fail(ar.cause());
+        });
+
+        return viewFuture;
+    }
+
+    private Future<AggregatedProfileInfo> loadProfile(AggregatedProfileNamingStrategy profileName) {
+        Future<AggregatedProfileInfo> future = doAsync(() -> profileLoader.load(profileName));
+
+        cache.put(profileName, future);
+        future.setHandler(ar -> {
+            logger.info("Profile load complete. file: {}", profileName);
+            // load_profile might fail, regardless reinsert to take the new utilization into account.
+            cache.put(profileName, future);
```

Comment:
earlier cache put will not fail. ar might have failed. I am reinserting irrespective of the completion status of ar.
Though this have changed now. 
Now reinserting in case we want to persist the nature of completion (success, deserialization error).
and ignoring (cache.invalidate & logging) in case of intermittent issue failures.